orchagent/nhgorch.cpp

```diff
@@ -0,0 +1,1386 @@
+     */
+    std::map<NextHopKey, sai_object_id_t> syncingMembers;
+
+    for (const auto& nh_key : nh_keys)
+    {
+        SWSS_LOG_INFO("Checking if next hop %s is valid",
+                        nh_key.to_string().c_str());
+
+        NextHopGroupMember& nhgm = m_members.at(nh_key);
+
+        /* If the member is already synced, continue. */
+        if (nhgm.getGmId() != SAI_NULL_OBJECT_ID)
+        {
+            SWSS_LOG_INFO("Update member");
+            continue;
+        }
+
+        /*
+         * If the next hop doesn't exist, stop from syncing the members.
+         */
+        if (nhgm.getNhId() == SAI_NULL_OBJECT_ID)
+        {
+            SWSS_LOG_WARN("Failed to get next hop %s in group %s",
+                        nhgm.to_string().c_str(), to_string().c_str());
+            return false;
+        }
+
+        /* If the neighbor's interface is down, skip from being syncd. */
+        if (gNeighOrch->isNextHopFlagSet(nh_key, NHFLAGS_IFDOWN))
+        {
+            SWSS_LOG_WARN("Skip next hop %s in group %s, interface is down",
+                        nh_key.to_string().c_str(), to_string().c_str());
+            continue;
+        }
+
+        /* Create the next hop group member's attributes and fill them. */
+        vector<sai_attribute_t> nhgm_attrs = createNhgmAttrs(nhgm);
+
+        /* Add a bulker entry for this member. */
+        nextHopGroupMemberBulker.create_entry(&syncingMembers[nh_key],
+                                               (uint32_t)nhgm_attrs.size(),
+                                               nhgm_attrs.data());
+    }
+
+    /* Flush the bulker to perform the sync. */
+    nextHopGroupMemberBulker.flush();
+
+    /*
+     * Go through the synced members and increment the Crm ref count for the
+     * successful ones.
+     */
+    bool success = true;
+    for (const auto& mbr : syncingMembers)
+    {
+        SWSS_LOG_INFO("Checking next hop member %s has a valid SAI ID",
+                        mbr.first.to_string().c_str());
+
+        /* Check that the returned member ID is valid. */
+        if (mbr.second == SAI_NULL_OBJECT_ID)
+        {
+            SWSS_LOG_ERROR("Failed to create next hop group %s's member %s",
+                            m_key.to_string().c_str(), mbr.first.to_string().c_str());
+            success = false;
+        }
+        else
+        {
+            m_members.at(mbr.first).sync(mbr.second);
+        }
+    }
+
+    SWSS_LOG_INFO("Returning %d", success);
+
+    return success;
+}
+/*
+ * Purpose:     Desync the given group's members over the SAI API.
+ *
+ * Description: Go through the given members and desync them.
+ *
+ * Params:      IN  nh_keys - The next hop keys of the members to desync.
+ *
+ * Returns:     true, if the operation was successful;
+ *              false, otherwise
+ */
+bool NextHopGroup::desyncMembers(const std::set<NextHopKey>& nh_keys)
+{
+    SWSS_LOG_ENTER();
+    SWSS_LOG_INFO("Removing members of group %s",
```

Comment:
general comment, this statement seems to be within 80 chars, any reason to split into two lines.

Reply:
I gave this as a general comment. Swss we have lines ~100 chars. We can follow the same to have 100 chars

Reply:
Fixed up ones I've found